### PR TITLE
49762 - Added API Automation FTs for ESS End Point

### PR DESCRIPTION
**Pull request by Mayuresh-Gawde**
Hi all,

This PR consist of functional API automation of POS - ESS End point. 

Please review.

Regards,
Mayuresh Gawde

### CODE REVIEW

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/ExchangeSetApiService.cs [hunk 1]
```diff
@@ -15,9 +15,9 @@ public class ExchangeSetApiService : IExchangeSetApiService
         private readonly IAuthTokenProvider _authTokenProvider;
         private readonly ILogger<ExchangeSetApiService> _logger;
 
-        public ExchangeSetApiService(ILogger<ExchangeSetApiService> logger,
-                                     IOptions<ExchangeSetApiConfiguration> exchangeSetApiConfiguration,
-                                     IExchangeSetApiClient exchangeSetApiClient, IAuthTokenProvider authTokenProvider)
+        public ExchangeSetApiService(ILogger<ExchangeSetApiService>? logger,
+                                     IOptions<ExchangeSetApiConfiguration>? exchangeSetApiConfiguration,
+                                     IExchangeSetApiClient? exchangeSetApiClient, IAuthTokenProvider? authTokenProvider)
         {
```
Comment (ajasonca) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/ExchangeSetApiService.cs:21]: What's the point of making these parameters nullable if the constructor then uses null coalescing assignments to throw an argument null exception? Why not just leave these parameters as not nullable?

**Review by ajasonca — CHANGES_REQUESTED** [review]
A couple of comments to check for me please

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        private TestConfiguration config { get; set; }
+        private GetCatalogue getcat { get; set; }
+        private string EssJwtToken { get; set; }
+        private GetProductIdentifiers getproductIdentifier { get; set; }
+
+        static EssAuthorizationConfiguration ESSAuth = new TestConfiguration().EssAuthorizationConfig;
+        static FleetManagerB2BApiConfiguration fleet = new TestConfiguration().fleetManagerB2BConfig;
+
+        List<string> productIdentifiers = new();
+        dynamic unpResponse;
```
[the remark this answers is not in the document]
Comment (Mayuresh-Gawde) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs]: Hi Jason,

We have updated the unpResponse to HttpResponseMessage.

Thanks.

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/ExchangeSetApiService.cs (hunk 1 above)
Comment (sumod13943) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/ExchangeSetApiService.cs]: Resolved. Removed the nullable parameters.

**Review by ajasonca — APPROVED** [review]

**Pull request merged**